kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/ZooKeeperUtils.java

```diff
@@ -26,38 +28,43 @@
  */
 @Slf4j
 public class ZooKeeperUtils {
-    public static void createPath(ZooKeeper zooKeeper, String zkPath, String subPath, byte[] data) {
+
+    private static boolean tryCreatePath(ZooKeeper zooKeeper, String path, byte[] data) {
         try {
-            if (zooKeeper.exists(zkPath, false) == null) {
-                zooKeeper.create(zkPath,
-                        new byte[0], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-            }
-            String addSubPath = zkPath + subPath;
-            if (zooKeeper.exists(addSubPath, false) == null) {
-                zooKeeper.create(addSubPath,
-                        data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-            } else {
-                zooKeeper.setData(addSubPath, data, -1);
+            zooKeeper.create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+        } catch (KeeperException e) {
+            if (!e.code().equals(KeeperException.Code.NODEEXISTS)) {
+                log.error("Failed to create ZooKeeper node {}: {}", path, e.getMessage());
+                return false;
             }
+        } catch (InterruptedException e) {
+            log.error("Failed to create ZooKeeper node {}: {}", path, e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+    public static void createPath(ZooKeeper zooKeeper, String zkPath, String subPath, byte[] data) {
+        tryCreatePath(zooKeeper, zkPath, new byte[0]);
```

Comment:
I think we can create znode for `groupIdStoredPath` at the time of KoP started instead of called every time.

Reply:
Done